public/app/features/scopes/selector/ScopesSelectorService.ts

```diff
@@ -233,6 +242,7 @@ export class ScopesSelectorService extends ScopesServiceBase<ScopesSelectorServi
     this.updateState({ selectedScopes: newSelectedScopes });
   };
 
+  // TODO: We should split this into two functions: expandNode and filterNode.
```

Comment:
something for this work or a followup?

Reply:
Not for this specific work, just a followup because it makes the logic very confusing